uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/TOILRequest.php

```diff
@@ -123,20 +125,44 @@ private static function validateTOILAmountIsValid($params) {
   }
 
   /**
-   * Validate that the TOIL amount is not greater than the maximum defined(if any) for
-   * the Absence Type
+   * Validate that the TOIL amount to be accrued plus total approved accrued TOIL for the period
+   * is not greater than the maximum defined(if any) for the Absence Type
    *
    * @param array $params
    *   The params array received by the create method
    *
    * @throws \CRM_HRLeaveAndAbsences_Exception_InvalidTOILRequestException
    */
   private static function validateValidTOILAmountNotGreaterThanMaximum($params) {
+    if(empty($params['contact_id']) || empty($params['type_id'])){
+      return;
+    }
+
     $absenceType = AbsenceType::findById($params['type_id']);
     $unlimitedAccrual = empty($absenceType->max_leave_accrual) && $absenceType->max_leave_accrual != 0;
-    if ($params['toil_to_accrue'] > $absenceType->max_leave_accrual && !$unlimitedAccrual) {
+
+    $currentPeriod = AbsencePeriod::getCurrentPeriod();
+    $startDate = new DateTime($currentPeriod->start_date);
+    $endDate = new DateTime($currentPeriod->end_date);
+
+    $leaveRequestStatuses = array_flip(LeaveRequest::buildOptions('status_id'));
+
+    $leaveRequestStatusFilter = [
+      $leaveRequestStatuses['Approved'],
+      $leaveRequestStatuses['Admin Approved']
+    ];
+
+    $totalApprovedTOIL = LeaveBalanceChange::getTotalTOILBalanceChangeForContact(
+      $params['contact_id'],
+      $startDate,
+      $endDate,
+      $leaveRequestStatusFilter
+    );
+    $totalProjectedToilForPeriod = $totalApprovedTOIL + $params['toil_to_accrue'];
```

Comment:
I would extract all this logic to get the toil balance for a period to another method. Then this validation method would only be something like:

```php
$totalProjectedToilForPeriod = $this->getTotalAppovedToilForPeriod() + $params['toil_to_accrued']
```